app/models/service_template_ansible_playbook.rb

```diff
@@ -7,6 +7,11 @@ def self.default_retirement_entry_point
     '/Service/Generic/StateMachines/GenericLifecycle/retire'
   end
 
+  def job_template(service_action)
```

Comment:
You can use `find_by` as shown https://github.com/ManageIQ/manageiq/blob/master/app/models/service_ansible_playbook.rb#L41
And you can change `ServiceAnsiblePlaybook` to delegate this method through the service template.